In [1]:
from scipy import stats
import numpy as np

# Question 1

5명의 학생들이 근무시간이 시작되면 조교님과 이야기하기 위해 기다리고 있습니다.
조교님은 한 번에 한 명씩 첫 번째 학생부터 다섯 번째 학생까지 상담을 합니다.
조교님이 학생과 대화하는 데 걸리는 시간이 평균 8분, 표준 편차가 2분인 정규 분포를 가지고 있다고 가정하고, 학생이 사용하는 시간이 서로 독립적이라고 가정합니다.

(a) 조교님이 5명의 학생과 대화하는 데 걸린 총 시간이 45분 이상일 확률은 얼마입니까?
(b) 조교님이 건강이 좋지 않은가 봅니다. 첫 번째 학생에게 말하기 시작한 이후에 조금 시간이 지나 두통이 생깁니다.
첫 학생의 상담 시작과 두통이 생기기 시작하는 시간 사이에 경과하는 시간이 평균 28분, 표준편차가 5분인 정규분포라고 가정해 봅시다
TA가 세 번째 학생과 대화를 마친 후에 두통이 시작될 확률은 얼마나 됩니까?

## Question 1 (a)

$$
X = \text{학생 한 명과 대화하는 데 걸리는 시간} \\
Y = 5X = \text{학생 5명과 대화하는데 걸리는 시간} \\
\mu = 8, \sigma = 2 \\
X \thicksim N(8, 2^{2}) \\
E(Y) = E(5X) = 5E(X) \\
Var(Y) = Var(5X) = 25Var(X) \\
\mu_{X} = 8, \sigma_{X} = 2 \\
\mu_{Y} = 40, \sigma_{Y} = 5 * 2 = 10 \\
P(Y \geq 45) \\
$$

따라서 Y = 45일 때 cdf 값을 구하고 1 - cdf하면 학생 5명과 대화를 할 때 걸린 총 시간이 45분 이상일 확률이 구해진다.

In [2]:
mu = 8
std = 2

answer_1_a = 1 - stats.norm.cdf(45, mu * 5, std * 2)
answer_1_a

0.10564977366685535

$$
P(Y \geq 45) = 0.002796018584074135
$$

## Question 1 (b)

$$
H = \text{첫 학생의 상담 시작과 두통이 생기기 시작하는 시간 사이의 경과하는 시간} \\
\mu_{H} = 28, \sigma_{H} = 5 \\
Y_{3} = 3X = \text{학생 3명과 대화하는데 걸리는 시간} \\
\mu_{Y_{3}} = 8 * 3 = 24, \sigma_{Y_{3}} = 2 * 3 = 6 \\
T = H - Y_{3}\\
\mu_{T} = 28 - 24 = 4, \sigma_{T} = 5 + 6 = 11 \\
P(T \geq 0)
$$

In [3]:
mu_t = 3
sigma_t = 11

answer_1_b = 1 - stats.norm.cdf(0, mu_t, sigma_t)
answer_1_b

0.6074685657262202

$$
P(T \geq 0) = 0.6074685657262202
$$

# Question 2

아래의 확률을 정규분포 근사로 계산해보고, 이항분포로 계산해본 후 비교해보세요.
B(n, p)는 n개 샘플, p 확률의 이항 분포라는 뜻입니다.
(a) P(X ≥ 7) where X ∼ B(10, 0.3)
(b) P(9 ≤ X ≤ 12) where X ∼ B(21, 0.5)
(c) P(X ≤ 3) where X ∼ B(7, 0.2)
(d) P(9 ≤ X ≤ 11) where X ∼ B(12, 0.65)

## 이항분포의 가우시안 근사

$$
B(n, p) \to N(np, np(1 - p)) \\
X \thicksim B(n, p) \\
\Rightarrow P(X \leq x) \simeq \Phi{(\frac{x + 0.5 - np}{\sqrt{np(1 - p)}})}\, and\, P(X \geq x) \simeq 1 - \Phi{(\frac{x - 0.5 - np}{\sqrt{np(1 - p)}})} \\
np \geq 5\, and\, n(1 - p) \geq 5\, \text{에서 잘 작동}
$$


## Question 2 (a)

$P(X \geq 7)\, where\, X \thicksim B(10, 0.3)$

In [4]:
n = 10
p = 0.3
q = 1 - p

mu = n * p
var = n * p * q

answer_2_a_norm = 1 - stats.norm.cdf((7 - 0.5 - mu) / np.sqrt(var))
answer_2_a_bi = 1 - stats.binom.cdf(7, n, p)

answer_2_a_norm, answer_2_a_bi

(0.007862649877252714, 0.0015903863999999768)

정규분포 근사값은 0.6567208193227195이고, 이항분포 값은 0.0015903863999999768이다.

## Question 2 (b)
$P(9 \leq X \leq 12)\, where\, X \thicksim B(21, 0.5)$

In [5]:
n = 21
p = 0.5
q = 1 - p

mu = n * p
var = n * p * q

answer_2_b_norm = stats.norm.cdf((12 + 0.5 - mu) / np.sqrt(var), 0, np.sqrt(var)) - stats.norm.cdf((9 - 0.5 - mu) / np.sqrt(var))
answer_2_b_bi = stats.binom.cdf(12, n, p) - stats.binom.cdf(9, n, p)

answer_2_b_norm, answer_2_b_bi

(0.4570141638579211, 0.4765329360961914)

정규분포 근사값은 0.4570141638579211이고, 이항분포 값은 0.4765329360961914이다.

## Question 2 (c)

$P(X \leq 3)\, where\, X \thicksim B(7, 0.2)$

In [6]:
n = 7
p = 0.2
q = 1 - p

mu = n * p
var = n * p * q

answer_2_c_norm = stats.norm.cdf((3 + 0.5 - mu) / np.sqrt(var))
answer_2_c_bi = stats.binom.cdf(3, n, p)

answer_2_c_norm, answer_2_c_bi

(0.9763895479982114, 0.966656)

정규분포 근사값은 0.9763895479982114이고, 이항분포 값은 0.966656이다.

## Question 2 (d)

$P(9 \leq X \leq 11)\, where\, X \thicksim B(12, 0.65)$

In [7]:
n = 12
p = 0.65
q = 1 - p

mu = n * p
var = n * p * q

answer_2_d_norm = stats.norm.cdf((11 + 0.5 - mu) / np.sqrt(var)) - stats.norm.cdf((9 + 0.5 - mu) / np.sqrt(var))
answer_2_d_bi = stats.binom.cdf(11, n, p) - stats.binom.cdf(9, n, p)

answer_2_d_norm, answer_2_d_bi

(0.1391995944105242, 0.14559956927216167)

정규분포 근사값은 0.1391995944105242이고, 이항분포 값은 0.14559956927216167이다.

# Question 3

한 연구자가 박테리아균을 배양합니다
하루 성장 후 배양물의 크기가 모수 μ = 2.3 및 σ = 0.2인 로그정규 분포를 갖는다고 가정합니다.
(a) 하루 후에 예상되는 배양물의 평균값은 얼마인가요?
(b) 하루 후의 배양물의 중간값은  얼마인가요?
(c) 하루 후의 배양물 크기의 3rd Quartile은 얼마인가요?
(d) 하루 후에 배양물 크기가 15보다 클 확률은 얼마인가요?
(e) 하루 후에 배양물 크기가 6보다 작을 확률은 얼마인가요?

$$
\mu = 2.3, \sigma = 0.2 \\
X = \text{하루 성장 후 배양물의 크기} \thicksim LogN(2.3, 0.2^{2})
$$

## Question 3 (a)

(a) 하루 후에 예상되는 배양물의 평균값은 얼마인가요?

log-normal distribution에서 평균은 $exp(\mu + \frac{\sigma^{2}}{2})$이다.

In [8]:
mu = 2.3
sigma = 0.2

answer_3_a = np.exp(mu + (sigma ** 2) / 2)
answer_3_a

10.175674306073333

하루 후 예상되는 배양물의 크기 평균 값 = 10.175674306073333

## Question 3 (b)

(b) 하루 후의 배양물의 중간값은  얼마인가요?

log-normal distribution에서 중간값은 $exp(\mu)$이다.

In [9]:
mu = 2.3
sigma = 0.2

answer_3_b = np.exp(mu)

answer_3_b

9.974182454814718

하루 후 예상되는 배양물의 크기 중간 값 = 9.974182454814718

## Question 3 (c)

(c) 하루 후의 배양물 크기의 3rd Quartile은 얼마인가요?

log-normal distribution에서 quantile은 $exp(\mu + \sqrt{2\sigma^{2}}\, erf^{-1}(2p - 1))$이다.

따라서 p가 0.75일 때 3rd Quartile이다.

In [10]:
from scipy import special

answer_3_c = np.exp(mu + np.sqrt(2 * (sigma ** 2)) / special.erf(2 * 0.75 - 1))

answer_3_c

17.174159769935745

하루 후 예상되는 배양물의 크기 3rd Quartile = 17.174159769935745

## Question 3 (d)

(d) 하루 후에 배양물 크기가 15보다 클 확률은 얼마인가요?

$P(X > 15) = 1 - \Phi{(\frac{ln(15) - \mu}{\sigma})}$

In [11]:
answer_3_d = 1 - stats.norm.cdf((np.log(15) - mu) / sigma)

answer_3_d

0.020662665969527705

$P(X > 15) = 0.020662665969527705$

## Question 4 (e)

(e) 하루 후에 배양물 크기가 6보다 작을 확률은 얼마인가요?

$P(X \leq 6) = \Phi{(\frac{ln(6) - \mu}{\sigma})}$

In [12]:
answer_3_e = stats.norm.cdf((np.log(6) - mu) / sigma)

answer_3_e

0.005523593376350241

$P(X \leq 6) = 0.005523593376350241$